File: src/AppBundle/Entity/Expense.php
<?php

namespace AppBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Expense
 *
 * @ORM\Table(name="expense")
 * @ORM\Entity(repositoryClass="AppBundle\Repository\ExpenseRepository")
 */
class Expense
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * 
     * @var \DateTime
     * 
     * @ORM\Column(name="expense_date", type="datetime")
     * @Assert\DateTime()
     */
    private $expenseDate;
    
    /**
     * @var string
     *
     * @ORM\Column(name="category", type="string", length=191)
     */
    private $category;

    /**
     * @var string
     *
     * @ORM\Column(name="employee_name", type="string", length=191)
     */
    private $employeeName;

    /**
     * @var string
     *
     * @ORM\Column(name="employee_address", type="text")
     */
    private $employeeAddress;

    /**
     * @var string
     *
     * @ORM\Column(name="expense_description", type="string", length=255)
     */
    private $expenseDescription;

    /**
     * @var string
     *
     * @ORM\Column(name="pre_tax_amount", type="decimal", precision=14, scale=4)
     */
    private $preTaxAmount;

    /**
     * @var string
     *
     * @ORM\Column(name="tax_name", type="string", length=255)
     */
    private $taxName;

    /**
     * @var string
     *
     * @ORM\Column(name="tax_amount", type="decimal", precision=14, scale=4)
     */
    private $taxAmount;

    /**
     * 
     * @var string
     * 
     * @ORM\Column(name="hash", type="string", length=40, unique=true)
     */
    private $hash;
    
    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set category
     *
     * @param string $category
     * @return Expense
     */
    public function setCategory($category)
    {
        $this->category = $category;

        return $this;
    }

    /**
     * Get category
     *
     * @return string 
     */
    public function getCategory()
    {
        return $this->category;
    }

    /**
     * Set employeeName
     *
     * @param string $employeeName
     * @return Expense
     */
    public function setEmployeeName($employeeName)
    {
        $this->employeeName = $employeeName;

        return $this;
    }

    /**
     * Get employeeName
     *
     * @return string 
     */
    public function getEmployeeName()
    {
        return $this->employeeName;
    }

    /**
     * Set employeeAddress
     *
     * @param string $employeeAddress
     * @return Expense
     */
    public function setEmployeeAddress($employeeAddress)
    {
        $this->employeeAddress = $employeeAddress;

        return $this;
    }

    /**
     * Get employeeAddress
     *
     * @return string 
     */
    public function getEmployeeAddress()
    {
        return $this->employeeAddress;
    }

    /**
     * Set expenseDescription
     *
     * @param string $expenseDescription
     * @return Expense
     */
    public function setExpenseDescription($expenseDescription)
    {
        $this->expenseDescription = $expenseDescription;

        return $this;
    }

    /**
     * Get expenseDescription
     *
     * @return string 
     */
    public function getExpenseDescription()
    {
        return $this->expenseDescription;
    }

    /**
     * Set preTaxAmount
     *
     * @param string $preTaxAmount
     * @return Expense
     */
    public function setPreTaxAmount($preTaxAmount)
    {
        $this->preTaxAmount = $preTaxAmount;

        return $this;
    }

    /**
     * Get preTaxAmount
     *
     * @return string 
     */
    public function getPreTaxAmount()
    {
        return $this->preTaxAmount;
    }

    /**
     * Set taxName
     *
     * @param string $taxName
     * @return Expense
     */
    public function setTaxName($taxName)
    {
        $this->taxName = $taxName;

        return $this;
    }

    /**
     * Get taxName
     *
     * @return string 
     */
    public function getTaxName()
    {
        return $this->taxName;
    }

    /**
     * Set taxAmount
     *
     * @param string $taxAmount
     * @return Expense
     */
    public function setTaxAmount($taxAmount)
    {
        $this->taxAmount = $taxAmount;

        return $this;
    }

    /**
     * Get taxAmount
     *
     * @return string 
     */
    public function getTaxAmount()
    {
        return $this->taxAmount;
    }
	
	/**
	 *
	 * @return \DateTime
	 */
	public function getExpenseDate() 
	{
		return $this->expenseDate;
	}
	
	/**
	 *
	 * @param \DateTime $expenseDate        	
	 * @return Expense
	 */
	public function setExpenseDate(\DateTime $expenseDate) 
	{
		$this->expenseDate = $expenseDate;
		
		return $this;
	}
	
	/**
	 *
	 * @return the string
	 */
	public function getHash() 
	{
		return $this->hash;
	}
	
	/**
	 *
	 * @param
	 *        	$hash
	 * @return Expense
	 */
	public function setHash($hash) 
	{
		$this->hash = $hash;
		
		return $this;
	}
	
	
}

File: src/main/java/com/wave/models/Expense.java
package com.wave.models;

import org.apache.commons.csv.CSVRecord;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

@Entity
@Table(name = "expense")
public class Expense {
    static SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy");
    @NotNull
    long insertId;
    @NotNull
    Date expenseDate;
    @NotNull
    String category;
    @NotNull
    String employeeName;
    @NotNull
    String employeeAddress;
    @NotNull
    String expenseDescription;
    @NotNull
    BigDecimal preTaxAmount;
    @NotNull
    String taxName;
    @NotNull
    BigDecimal taxAmount;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    public Expense(long id, CSVRecord record) throws ParseException, NumberFormatException {
        insertId = id;
        expenseDate = format.parse(record.get(0));
        category = record.get(1);
        employeeName = record.get(2);

        employeeAddress = record.get(3);
        expenseDescription = record.get(4);
        preTaxAmount = new BigDecimal(record.get(5).trim().replace(",", ""));
        taxName = record.get(6);
        taxAmount = new BigDecimal(record.get(7).trim().replace(",", ""));
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getInsertId() {
        return insertId;
    }

    public void setInsertId(long insertId) {
        this.insertId = insertId;
    }

    public Date getExpenseDate() {
        return expenseDate;
    }

    public void setExpenseDate(Date expenseDate) {
        this.expenseDate = expenseDate;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getEmployeeName() {
        return employeeName;
    }

    public void setEmployeeName(String employeeName) {
        this.employeeName = employeeName;
    }

    public String getEmployeeAddress() {
        return employeeAddress;
    }

    public void setEmployeeAddress(String employeeAddress) {
        this.employeeAddress = employeeAddress;
    }

    public String getExpenseDescription() {
        return expenseDescription;
    }

    public void setExpenseDescription(String expenseDescription) {
        this.expenseDescription = expenseDescription;
    }

    public BigDecimal getPreTaxAmount() {
        return preTaxAmount;
    }

    public void setPreTaxAmount(BigDecimal preTaxAmount) {
        this.preTaxAmount = preTaxAmount;
    }

    public String getTaxName() {
        return taxName;
    }

    public void setTaxName(String taxName) {
        this.taxName = taxName;
    }

    public BigDecimal getTaxAmount() {
        return taxAmount;
    }

    public void setTaxAmount(BigDecimal taxAmount) {
        this.taxAmount = taxAmount;
    }

}
